doc/nrf/ug_nrf9160.rst

```diff
@@ -101,10 +162,59 @@ If the modem is shut down gracefully before the next boot (by using ``AT+CFUN=0`
+FOTA upgrades can be used to apply delta patches to the `LTE modem`_ firmware and to replace the upgradable bootloader or the application.
+
+.. note::
+   Even though the Secure Partition Manager and the application are two individually compiled components, they are treated as a single binary blob in the context of firmware upgrades.
+   When we refer to the application in this section, we therefore mean the application including the Secure Partition Manager.
+
+A FOTA upgrade requires the following steps:
+
+1. Make sure that your application supports FOTA upgrades.
+      To download and apply FOTA upgrades, your application must use the :ref:`lib_fota_download` library.
+      This library deduces the type of upgrade by inspecting the header of the firmware and invokes the :ref:`lib_dfu_target` library to apply the firmware upgrade.
+      By default, the DFU target library supports all kinds of FOTA upgrades, but you can disable support for specific targets.
+
+      In addition, the following requirements apply:
+
+      * If you want to upgrade the application, :doc:`mcuboot:index` must be used as upgradable bootloader (:option:`CONFIG_BOOTLOADER_MCUBOOT`).
+      * If you want to upgrade the upgradable bootloader, the :ref:`bootloader` must be used (:option:`CONFIG_SECURE_BOOT`).
+      * If you want to upgrade the modem firmware, neither MCUboot nor the immutable bootloader are required, because the modem firmware upgrade is handled by the modem itself.
+
+#. Create a binary file that contains the new image.
+      This step does not apply for upgrades of the modem firmware.
+      You can download delta patches for the modem firmware from the `nRF9160 product website (compatible downloads)`_.
+
+      To create a binary file for an application upgrade, make sure that (:option:`CONFIG_BOOTLOADER_MCUBOOT`) is enabled and build the application as usual.
+      The build will create several binary files (see :ref:`mcuboot:mcuboot_ncs`).
+      The :file:`app_update.bin` file is the file that should be uploaded to the server.
+
+      To create a binary file for a bootloader upgrade, make sure that (:option:`CONFIG_SECURE_BOOT`) is enabled and build MCUboot as usual.
+      The build will create a binary file that contains two variants of the upgradable bootloader, one for each bootloader slot.
+      See :ref:`upgradable_bootloader` for more information.
+
+#. Make the binary file available for download.
+     Upload the file to a web server that is compatible with the :ref:`lib_download_client` library.
+     One way of doing this is to upload the file to an Amazon Web Services Simple Storage Service (AWS S3) bucket.
+     See the :ref:`lib_aws_fota` documentation for instructions.
+
+     Your application must be able to retrieve the URL for the binary file.
+     This URL can be hardcoded in the application, or you can use functionality like AWS jobs to provide the URL dynamically.
```

Comment:
The URL will differ depending on what kind of update this is.

Reply:
But you can probably just hardcode the "file name" with two different paths?

Reply:
It is not a good idea to hardcode the URL in the application, but for development purposes it can be useful. If it is cumbersome to explain, you could simply drop the part about hard coding it in the application.

Reply:
Linking to FOTA download.